### PR TITLE
Ia connect session

### PR DESCRIPTION
**Pull request by LukeMcGovern-UL**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by LukeMcGovern-UL** [comment]
Hi Team,

Having compared the modifications in version 9.1, I believe the following have failed the Swagger and Change Validation checks:

- Output added to RunAlternativeIAConnectSentFromDirector action
_This is not a breaking change because there were previously no outputs returned_

- IAConnectSessionName output now mandatory for AttachToIAConnectSessionByName, DoesIAConnectSessionExistByName, WaitForIAConnectSessionToCloseByName and KillIAConnectSessionByName
_This is not a breaking change because this value was always required by our API and the action would fail if no value was provided_

- FilePath output now mandatory for GetFileSecurityPermissions
_This is not a breaking change because this value was always required by our API and the action would fail if no value was provided_

File: certified-connectors/IA-Connect Session/apiDefinition.swagger.json [hunk 1]
```diff
@@ -4173,6 +4185,658 @@
...
+                "CredentialType",
+                "Workflow"
+              ]
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {}
```
Comment (sriyen-msft) [inline certified-connectors/IA-Connect Session/apiDefinition.swagger.json:4605]: @LukeMcGovern-UL Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/IA-Connect Session/apiDefinition.swagger.json [hunk 2]
```diff
@@ -4173,6 +4185,658 @@
...
```
Comment (sriyen-msft) [inline certified-connectors/IA-Connect Session/apiDefinition.swagger.json:4459]: @LukeMcGovern-UL Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/IA-Connect Session/apiDefinition.swagger.json [hunk 3]
```diff
@@ -4173,6 +4185,658 @@
...
+                "RDPFilePath",
+                "Workflow"
+              ]
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {}
```
Comment (sriyen-msft) [inline certified-connectors/IA-Connect Session/apiDefinition.swagger.json:4834]: @LukeMcGovern-UL Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/IA-Connect Session/apiDefinition.swagger.json [hunk 4]
```diff
@@ -7724,6 +8408,79 @@
...
+                "CSVFilePath",
+                "Workflow"
+              ]
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {}
```
Comment (sriyen-msft) [inline certified-connectors/IA-Connect Session/apiDefinition.swagger.json:8478]: @LukeMcGovern-UL Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: certified-connectors/IA-Connect Session/apiDefinition.swagger.json (hunk 1 above)
Comment (LukeMcGovern-UL) [reply certified-connectors/IA-Connect Session/apiDefinition.swagger.json:4605]: @sriyen-msft Thank you for reviewing our pull request - in this scenario the command returns no output and therefore has no output schema. We distinguish between success or failure by returning the relevant HTML code, 200 (success) or 400 (failure) which will result in an exception in the Power Automate Flow. 

Most of our actions do return data and all of these have an appropriate output schema defined.

File: certified-connectors/IA-Connect Session/apiDefinition.swagger.json (hunk 2 above)
Comment (LukeMcGovern-UL) [reply certified-connectors/IA-Connect Session/apiDefinition.swagger.json:4459]: @sriyen-msft Thank you for reviewing our pull request - in this scenario the command returns no output and therefore has no output schema. We distinguish between success or failure by returning the relevant HTML code, 200 (success) or 400 (failure) which will result in an exception in the Power Automate Flow. 

Most of our actions do return data and all of these have an appropriate output schema defined.

File: certified-connectors/IA-Connect Session/apiDefinition.swagger.json (hunk 3 above)
Comment (LukeMcGovern-UL) [reply certified-connectors/IA-Connect Session/apiDefinition.swagger.json:4834]: @sriyen-msft Thank you for reviewing our pull request - in this scenario the command returns no output and therefore has no output schema. We distinguish between success or failure by returning the relevant HTML code, 200 (success) or 400 (failure) which will result in an exception in the Power Automate Flow. 

Most of our actions do return data and all of these have an appropriate output schema defined.

File: certified-connectors/IA-Connect Session/apiDefinition.swagger.json (hunk 4 above)
Comment (LukeMcGovern-UL) [reply certified-connectors/IA-Connect Session/apiDefinition.swagger.json:8478]: @sriyen-msft Thank you for reviewing our pull request - in this scenario the command returns no output and therefore has no output schema. We distinguish between success or failure by returning the relevant HTML code, 200 (success) or 400 (failure) which will result in an exception in the Power Automate Flow. 

Most of our actions do return data and all of these have an appropriate output schema defined.

File: certified-connectors/IA-Connect Session/apiDefinition.swagger.json (hunk 2 above)
Comment (sriyen-msft) [reply certified-connectors/IA-Connect Session/apiDefinition.swagger.json:4459]: @LukeMcGovern-UL The response code "success" should return something, at least a string with a message called "success" or "failed" which needs to be defined in the schema. An empty schema definition would not be allowed anymore.

Comment (LukeMcGovern-UL) [reply certified-connectors/IA-Connect Session/apiDefinition.swagger.json]: @sriyen-msft Ok - We have updated the Swagger definitions to return a boolean to show success or failure and any error messages in a separate string.

**Review by LukeMcGovern-UL — COMMENTED** [review]
Changes made as requested.

File: certified-connectors/IA-Connect Session/apiDefinition.swagger.json [hunk 5]
```diff
@@ -3927,68 +3939,591 @@
...
+              "required": [
+                "Workflow"
+              ]
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {}
```
Comment (sriyen-msft) [inline certified-connectors/IA-Connect Session/apiDefinition.swagger.json:4022]: @LukeMcGovern-UL Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/IA-Connect Session/apiDefinition.swagger.json [hunk 6]
```diff
@@ -3927,68 +3939,591 @@
...
```
Comment (sriyen-msft) [inline certified-connectors/IA-Connect Session/apiDefinition.swagger.json:4064]: @LukeMcGovern-UL  Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/IA-Connect Session/apiDefinition.swagger.json [hunk 7]
```diff
@@ -3927,68 +3939,591 @@
...
+                "Filepath",
+                "Workflow"
+              ]
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {}
```
Comment (sriyen-msft) [inline certified-connectors/IA-Connect Session/apiDefinition.swagger.json:4118]: @LukeMcGovern-UL Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: certified-connectors/IA-Connect Session/apiDefinition.swagger.json (hunk 5 above)
Comment (LukeMcGovern-UL) [reply certified-connectors/IA-Connect Session/apiDefinition.swagger.json]: Updated output schema as requested.

File: certified-connectors/IA-Connect Session/apiDefinition.swagger.json (hunk 7 above)
Comment (LukeMcGovern-UL) [reply certified-connectors/IA-Connect Session/apiDefinition.swagger.json]: Updated output schema as requested.

File: certified-connectors/IA-Connect Session/apiDefinition.swagger.json (hunk 6 above)
Comment (LukeMcGovern-UL) [reply certified-connectors/IA-Connect Session/apiDefinition.swagger.json]: Updated output schema as requested.

**Review by sriyen-msft — APPROVED** [review]

**Pull request merged**